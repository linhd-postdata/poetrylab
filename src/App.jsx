import React, { Component } from 'react'

import Main from './components/Main'
import AddForm from './components/AddForm'


class App extends Component {
  state = {
    poems: [],
    current: -1
  }

  constructor(props) {
    super(props);
    this.addPoem = this.addPoem.bind(this);
    this.changeCurrentPoem = this.changeCurrentPoem.bind(this);
    this.updatePoem = this.updatePoem.bind(this);
  }

  addPoem(title, text) {
    this.setState({
      poems: [...this.state.poems, {title, text, analysis: {}}],
      current: this.state.poems.length,
    })
  }

  updatePoem(poemProps) {
    const { poems, current } = this.state;
    poems[current] = poemProps;
    this.setState({ poems });
    this.forceUpdate();
  }

  changeCurrentPoem(index) {
    this.setState({
      current: index,
    })
  }

  render() {
    return (
      <div>
        <Main {...this.state}
          changeCurrent={this.changeCurrentPoem}
          updatePoem={this.updatePoem}
        />
        <AddForm onSubmit={this.addPoem} />
      </div>
    );
  }
}
export default App;