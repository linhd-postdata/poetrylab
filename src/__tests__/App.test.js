// Link.react.test.js
import React from 'react';
import App from '../App';
import { create } from 'react-test-renderer';

test('App renders properly', () => {
  const component = create(
    <App />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

//   // manually trigger the callback
//   tree.props.onMouseEnter();
//   // re-rendering
//   tree = component.toJSON();
//   expect(tree).toMatchSnapshot();

//   // manually trigger the callback
//   tree.props.onMouseLeave();
//   // re-rendering
//   tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
});