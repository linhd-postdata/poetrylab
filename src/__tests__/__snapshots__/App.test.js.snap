// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
<div>
  <div
    className="Main-root-1"
  >
    <header
      className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed Main-appBar-2 MuiAppBar-colorPrimary mui-fixed"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular"
      >
        <button
          aria-label="Open drawer"
          className="MuiButtonBase-root MuiIconButton-root Main-menuButton-4 MuiIconButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit MuiTypography-noWrap"
        >
          PoetryLab: 
          
        </h6>
      </div>
    </header>
    <div
      className="MuiDrawer-root MuiDrawer-docked Main-drawer-6"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation0 MuiDrawer-paper Main-drawerPaper-7 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft"
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <div
          className="Main-drawerHeader-8"
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </span>
          </button>
        </div>
        <hr
          className="MuiDivider-root"
        />
        <ul
          className="MuiList-root MuiList-padding"
        />
        <hr
          className="MuiDivider-root"
        />
        <ul
          className="MuiList-root MuiList-padding"
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.5 6v11.5c0 2.21-1.79 4-4 4s-4-1.79-4-4V5c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5v10.5c0 .55-.45 1-1 1s-1-.45-1-1V6H10v9.5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V5c0-2.21-1.79-4-4-4S7 2.79 7 5v12.5c0 3.04 2.46 5.5 5.5 5.5s5.5-2.46 5.5-5.5V6h-1.5z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Upload a zip
              </span>
            </div>
          </div>
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM17 13l-5 5-5-5h3V9h4v4h3z"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Download all
              </span>
            </div>
          </div>
        </ul>
      </div>
    </div>
    <main
      className="Main-content-9"
    >
      <div
        className="Main-drawerHeader-8"
      />
      To start, add a poem
      
    </main>
  </div>
  <div>
    <button
      aria-label="Add"
      className="MuiButtonBase-root MuiFab-root AddForm-fab-143 MuiFab-extended MuiFab-primary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiFab-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
        Add Poem
      </span>
    </button>
  </div>
</div>
`;
